test/sql_tests.cpp

```diff
@@ -40,27 +43,75 @@ TEST(CreateStatementTest) {
+  ASSERT_EQ(stmt->columns->at(5)->type, (ColumnType{DataType::DECIMAL, 0, 13, 37}));
+  ASSERT_NEQ(stmt->columns->at(5)->type, (ColumnType{DataType::DECIMAL, 0, 14, 37}));
+  ASSERT_NEQ(stmt->columns->at(5)->type, (ColumnType{DataType::DECIMAL, 0, 13, 38}));
+  ASSERT_EQ(stmt->columns->at(6)->type, (ColumnType{DataType::DOUBLE}));
+  ASSERT_EQ(stmt->columns->at(7)->type, (ColumnType{DataType::FLOAT}));
+  ASSERT_EQ(stmt->columns->at(8)->type, (ColumnType{DataType::INT}));
+  ASSERT_EQ(stmt->columns->at(9)->type, (ColumnType{DataType::INT}));
+  ASSERT_EQ(stmt->columns->at(10)->type, (ColumnType{DataType::LONG}));
+  ASSERT_EQ(stmt->columns->at(11)->type, (ColumnType{DataType::REAL}));
+  ASSERT_EQ(stmt->columns->at(12)->type, (ColumnType{DataType::SMALLINT}));
+  ASSERT_EQ(stmt->columns->at(13)->type, (ColumnType{DataType::TEXT}));
+  ASSERT_EQ(stmt->columns->at(14)->type, (ColumnType{DataType::TIME}));
+  ASSERT_EQ(stmt->columns->at(15)->type, (ColumnType{DataType::TIME, 0, 17}));
+  ASSERT_NEQ(stmt->columns->at(15)->type, (ColumnType{DataType::TIME, 0, 18}));
+  ASSERT_EQ(stmt->columns->at(16)->type, (ColumnType{DataType::VARCHAR, 50}));
+  ASSERT_NEQ(stmt->columns->at(16)->type, (ColumnType{DataType::VARCHAR, 51}));
+  ASSERT_EQ(stmt->columns->at(17)->type, (ColumnType{DataType::VARCHAR, 60}));
+  ASSERT_NEQ(stmt->columns->at(17)->type, (ColumnType{DataType::VARCHAR, 61}));
+
   ASSERT_EQ(stmt->columns->at(0)->nullable, false);
   ASSERT_EQ(stmt->columns->at(1)->nullable, false);
-  ASSERT_EQ(stmt->columns->at(2)->nullable, true);
+  ASSERT_EQ(stmt->columns->at(2)->nullable, false);
   ASSERT_EQ(stmt->columns->at(3)->nullable, false);
   ASSERT_EQ(stmt->columns->at(4)->nullable, false);
-  ASSERT_EQ(stmt->columns->at(4)->column_constraints->size(), 2);
-  ASSERT_STREQ(stmt->tableConstraints->at(0)->columnNames->at(0), "name");
-  ASSERT_STREQ(stmt->tableConstraints->at(0)->columnNames->at(1), "student_number");
+  ASSERT_EQ(stmt->columns->at(5)->nullable, false);
+  ASSERT_EQ(stmt->columns->at(6)->nullable, false);
+  ASSERT_EQ(stmt->columns->at(7)->nullable, false);
+  ASSERT_EQ(stmt->columns->at(8)->nullable, false);
+  ASSERT_EQ(stmt->columns->at(9)->nullable, true);
+  ASSERT_EQ(stmt->columns->at(10)->nullable, false);
+  ASSERT_EQ(stmt->columns->at(11)->nullable, false);
+  ASSERT_EQ(stmt->columns->at(12)->nullable, false);
+  ASSERT_EQ(stmt->columns->at(13)->nullable, false);
+  ASSERT_EQ(stmt->columns->at(14)->nullable, false);
+  ASSERT_EQ(stmt->columns->at(15)->nullable, false);
+  ASSERT_EQ(stmt->columns->at(16)->nullable, false);
+  // Expecting two column constraints since the NULL constraint gets explicitly removed and the corresponding
+  // information is then stored in `ColumnDefinition::nullable` instead.
+  ASSERT_EQ(stmt->columns->at(13)->column_constraints->size(), 2);
+  ASSERT_STREQ(stmt->tableConstraints->at(0)->columnNames->at(0), "c_char");
```

Comment:
Add a comment to explain the table constraint in the example?
//```PRIMARY KEY(c_char, c_int)``` is a/the table constraint.
//Table constraints are identified and separated during the parsing of the SQL string